packages/core/src/indexing-store/cache.ts

```diff
@@ -646,6 +646,7 @@ export const createIndexingCache = ({
                 .map(({ sql }) => `target."${sql}" = source."${sql}"`)
                 .join(" AND ")};
             `;
+          const truncateQuery = `TRUNCATE TABLE "${getTableName(table)}" CASCADE`;
```

Comment:
I'd probably opt for dropping the temp table rather than create if not exists and truncate. That way the flush function doesn't leave around any state in the transaction. Not a strong opinion though.